File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/validator/TypeConverterValidator.java
package com.raizlabs.android.dbflow.processor.validator;

import com.raizlabs.android.dbflow.processor.definition.TypeConverterDefinition;
import com.raizlabs.android.dbflow.processor.model.ProcessorManager;

/**
 * Description:
 */
public class TypeConverterValidator implements Validator<TypeConverterDefinition> {
    @Override
    public boolean validate(ProcessorManager processorManager, TypeConverterDefinition typeConverterDefinition) {
        boolean success = true;

        if (typeConverterDefinition.getModelTypeName() == null) {
            processorManager.logError("TypeConverter: " + typeConverterDefinition.getClassName().toString() +
                    " uses an unsupported Model Element parameter. If it has type parameters, you must remove them or subclass it" +
                    "for proper usage.");
            success = false;
        } else if (typeConverterDefinition.getDbTypeName() == null) {
            processorManager.logError("TypeConverter: " + typeConverterDefinition.getClassName().toString() +
                    " uses an unsupported DB Element parameter. If it has type parameters, you must remove them or subclass it " +
                    "for proper usage.");
            success = false;
        }

        return success;
    }
}

File: dbflow-processor/src/main/java/com/raizlabs/android/dbflow/processor/validator/TypeConverterValidator.kt
package com.raizlabs.android.dbflow.processor.validator

import com.raizlabs.android.dbflow.processor.definition.TypeConverterDefinition
import com.raizlabs.android.dbflow.processor.model.ProcessorManager

/**
 * Description:
 */
class TypeConverterValidator : Validator<TypeConverterDefinition> {
    override fun validate(processorManager: ProcessorManager,
                          typeConverterDefinition: TypeConverterDefinition): Boolean {
        var success = true

        if (typeConverterDefinition.modelTypeName == null) {
            processorManager.logError("TypeConverter: " + typeConverterDefinition.className.toString() +
                    " uses an unsupported Model Element parameter. If it has type parameters, you must remove them or subclass it" +
                    "for proper usage.")
            success = false
        } else if (typeConverterDefinition.dbTypeName == null) {
            processorManager.logError("TypeConverter: " + typeConverterDefinition.className.toString() +
                    " uses an unsupported DB Element parameter. If it has type parameters, you must remove them or subclass it " +
                    "for proper usage.")
            success = false
        }

        return success
    }
}
